source/daplink/drag-n-drop/flash_decoder.c

```diff
@@ -347,11 +355,19 @@ static bool flash_decoder_is_at_end(uint32_t addr, const uint8_t *data, uint32_t
         case FLASH_DECODER_TYPE_INTERFACE:
             end_addr = DAPLINK_ROM_IF_START + DAPLINK_ROM_IF_SIZE;
             break;
-        
+
         case FLASH_DECODER_TYPE_TARGET:
             //only if we are sure it is a bin for the target; without check unordered hex files will cause to terminate flashing
             if (flash_type_target_bin && g_board_info.target_cfg) {
+#ifdef TRUSTZONE_M
+                if(addr >= g_board_info.target_cfg->flash_start_alias) {
+                    end_addr = g_board_info.target_cfg->flash_end_alias;
+                } else {
+                    end_addr = g_board_info.target_cfg->flash_end;
+                }
+#else
```

Comment:
TODO:

Add patch for multi regions instead of this logic.

Reply:
Will remove it.